netatmo_influx.py

```diff
@@ -84,65 +96,75 @@ def parse_args():
         influx_username = environ.get("INFLUX_USERNAME")
     if environ.get("INFLUX_PASSWORD"):
         influx_password = environ.get("INFLUX_PASSWORD")
-
-    try:
-        authorization = lnetatmo.ClientAuth(
-            clientId=client_id,
-            clientSecret=client_secret,
-            username=netatmo_username,
-            password=netatmo_password,
-            scope="read_station",
-        )
-
-        weatherData = lnetatmo.WeatherStationData(authorization)
-        client = InfluxDBClient(
-            host=influx_host,
-            port=influx_port,
-            username=influx_username,
-            password=influx_password,
-            database=influx_db,
-            ssl=influx_ssl
-        )
-
-        for station in weatherData.stations:
-            station_data = []
-            module_data = []
-            station = weatherData.stationById(station)
-            station_name = station["station_name"]
-            altitude = station["place"]["altitude"]
-            country = station["place"]["country"]
-            timezone = station["place"]["timezone"]
-            longitude = station["place"]["location"][0]
-            latitude = station["place"]["location"][1]
-            for module, moduleData in weatherData.lastData(station=station_name, exclude=3600).items():
-                for measurement in ["altitude", "country", "longitude", "latitude", "timezone"]:
-                    value = eval(measurement)
-                    if type(value) == int:
-                        value = float(value)
-                    station_data.append(
-                        {
-                            "measurement": measurement,
-                            "tags": {"station": station_name, "module": module},
-                            "time": moduleData["When"],
-                            "fields": {"value": value},
-                        }
-                    )
-
-                for sensor, value in moduleData.items():
-                    if sensor.lower() != "when":
+    if interval is None:
+        interval = 300  # interval in seconds; default are 5 Minutes
+    elif environ.get("INTERVAL"):
+        interval = int(environ.get("INTERVAL"))
+
+    while running:
+        signal.signal(signal.SIGTERM, shutdown)
```

Comment:
We don't have to do the linking with the signals several times, it is sufficient to do it when starting the main function.